docs/CONFIG-VARS.md

```diff
@@ -54,22 +54,26 @@ For recommendations on how to set these variables in your environment, see [Auth
-Contact your Network Administrator to find the public CIDR range of your network.
+By default, the public endpoints of the Azure resources that are being created
+are only accessible through authenticated Azure clients
+(such as the Azure Portal, the `az` CLI, the Azure Shell, etc.).
+To allow access for other administrative client applications (for example `kubectl`, `psql`, `ssh` etc.), you can set Network Security Group (NSG) rules to control access from your source IP addresses.
+
+To do set these permissions as part of this Terraform script, specify ranges of IP addresses in [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) with the following variables.
+
+NOTE: When deploying infrastructure into a private network (e.g. a VPN), with no public endpoints, the options documented in this block are not applicable.
+
+NOTE: The script will either create a new NSG, or use an existing NSG, if specified in the [`nsg_name`](#use-existing) variable. 
 
 You can use `default_public_access_cidrs` to set a default range for all created resources. To set different ranges for other resources, define the appropriate variable. Use an empty list `[]` to disallow access explicitly.
 
 | Name | Description | Type | Default | Notes |
 | :--- | ---: | ---: | ---: | ---: |
 | default_public_access_cidrs | IP address ranges allowed to access all created cloud resources | list of strings | | Sets a default for all resources. |
-| cluster_endpoint_public_access_cidrs | IP address ranges allowed to access the AKS cluster API | list of strings | | For client admin access to the cluster (by `kubectl`, for example). |
-| vm_public_access_cidrs | IP address ranges allowed to access the VMs | list of strings | | Opens port 22 for SSH access to the jump server and/or NFS VM by adding Ingress Rule on the NSG |
-| postgres_public_access_cidrs | IP address ranges allowed to access the Azure PostgreSQL Server | list of strings || Opens port 5432 by adding Ingress Rule on the NSG |
-| acr_public_access_cidrs | IP address ranges allowed to access the ACR instance | list of strings |||
+| cluster_endpoint_public_access_cidrs | IP address ranges allowed to access the AKS cluster API | list of strings | | For client admin access to the cluster api (by `kubectl`, for example). Only used with `cluster_api_mode=public`|
+| vm_public_access_cidrs | IP address ranges allowed to access the VMs | list of strings | | Opens port 22 for SSH access to the jump server and/or NFS VM by adding Ingress Rule on the NSG. Only used iwth `create_jump_public_ip=true` or `create_nfs_public_ip=true`   |
+| postgres_public_access_cidrs | IP address ranges allowed to access the Azure PostgreSQL Server | list of strings || Opens port 5432 by adding Ingress Rule on the NSG. Only used when creating postgres instances. |
+| acr_public_access_cidrs | IP address ranges allowed to access the ACR instance | list of strings || Only used with `create_container_registry=true` |
```

Comment:
Are there guards on these variables later in the code to keep a user from setting these when not needed?

Reply:
cluster_api: https://github.com/sassoftware/viya4-iac-azure/blob/private-cluster/locals.tf#L11
vms: https://github.com/sassoftware/viya4-iac-azure/blob/private-cluster/vms.tf#L118
postgres: https://github.com/sassoftware/viya4-iac-azure/blob/private-cluster/locals.tf#L23
acr: https://github.com/sassoftware/viya4-iac-azure/blob/private-cluster/main.tf#L118